### PR TITLE
remove reaper, ttl and heartbeat logic

### PR DESCRIPTION
**Pull request by ddadlani**
concourse/concourse#3495

Signed-off-by: Divya Dadlani <ddadlani@pivotal.io>
Co-authored-by: Krishna Mannem <kmannem@pivotal.io>

### CODE REVIEW

**Comment by cirocosta** [comment]
Hey, two questions that came up during standup, maybe not too specific to the PR: 

- was there a particular worker version when we used the reaper functionality?
- getting rid of the reaping functionality from baggageclaim, should we bump the `worker` version, or is the worker version more for signaling the compatibility with web's APIs and expectations (TSA more specifically)?

Thanks

**Comment by vito** [comment]
> * getting rid of the reaping functionality from baggageclaim, should we bump the `worker` version, or is the worker version more for signaling the compatibility with web's APIs and expectations (TSA more specifically).

The latter - we shouldn't need to bump the worker version for this since we've stopped using these TTL APIs a long time ago. The fields are also [marked `,omitempty`](https://github.com/concourse/baggageclaim/blob/8f1eaeab1832501f1488a8b966bbf80f4307d0ad/resources.go#L12) so we should no have problems related to older `web` nodes sending bogus fields.

Also I don't think we even had worker versions at the time, or maybe they were introduced as we transitioned away from TTLs. Either way it should be OK to remove.

(Looks like there are conflicts though.)

File: doc.go [hunk 1]
```diff
@@ -20,10 +20,8 @@ existing, and then dying. This state model is required as creating large
 amounts of data can potentially take a long time to materialize. You are only
 able to interact with volumes that are in the middle state.
 
-Volumes are given a TTL which means that BaggageClaim can reclaim the disk
-space used by the volumes over the life of the service. Parent-child
-relationships between volumes are understood and will prevent a parent being
-deleted while a child is still active.
+Parent-child relationships between volumes are understood and will prevent a
```
Comment (vito) [inline doc.go]: I don't think this is actually true - TTLs were the only way this was supported. I guess we should just remove the paragraph?

**Review by vito — CHANGES_REQUESTED** [review]

File: doc.go (hunk 1 above)
Comment (ddadlani) [reply doc.go]: @vito sure, can do. We do manage volume relationships on the ATC though. Is that documented anywhere? Should it be?

Comment (vito) [reply doc.go]: BaggageClaim doesn't know anything about the ATC, but if you want the docs to reflect that here we could say that it's the responsibility of the API consumer to delete child volumes before parent volumes.

**Comment by topherbullock** [comment]
>Also I don't think we even had worker versions at the time, or maybe they were introduced as we transitioned away from TTLs. Either way it should be OK to remove.

@vito yeah.. it was [a long long time ago that we removed the last reference to setting TTLs on Volumes ](https://github.com/concourse/atc/commit/5ad298f9cc7cbd94b086cf241898bfd43f218ef5#diff-b1bbf954376b20b078cb6b5e154e8a42)

File: doc.go (hunk 1 above)
Comment (ddadlani) [reply doc.go]: included.

**Review by vito — APPROVED** [review]

**Comment by vito** [comment]
I'll create a v2.0.0 tag after merging this.

**Comment by vito** [comment]
Oi, looks like a v2.0.0 tag would necessitate a lot of work:

https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher

Technically this is a major backwards-incompatible change, but the only client consumer that we know of is Concourse, so...maybe I'll just bump to v1.4.0. 😕

**Pull request merged**